spinalcordtoolbox/image.py

```diff
@@ -1704,3 +1707,66 @@ def compute_cross_corr_3d(image: Image, coord, xrange=list(range(-10, 10)), xshi
     # Change adjust rl_coord
     logger.info('R-L coordinate adjusted from %s to  %s)', x, x + xrange[ind_peak])
     return x + xrange[ind_peak]
+
+
+def stitch_images(fnames_in: list, fname_out='stitched.nii.gz'):
+    """
+    Stitch two (or more) images utilizing the C++-precompiled binaries of Biomedia-MIRA's stitching toolkit
+    (https://github.com/biomedia-mira/stitching) by placing a system call.
+
+    :param fnames_in: list of filenames of scans in RPI orientation. e.g. [stack1.nii.gz, stack2.nii.gz]
+    :param fname_out: filename for stitched output scan.
+    :param directory: path to temp directory where we store the result of stitching.nii.gz
+    :return: none
+    """
+    # stringify the fname list to parsable cmd parameter
+    fnames_cmd = " ".join(fnames_in)
+
+    cmd = ['isct_stitching', '-i', fnames_cmd, '-o', fname_out, '-a']
+    status, output = run_proc(cmd, verbose='verbose', is_sct_binary=True)
+    print(status, output)
+    if status != 0:
+        raise RuntimeError(f"Subprocess call {cmd} returned non-zero: {output}")
+
+    return Image(fname_out)
+
+
+def generate_stitched_qc_images(ims_in: Sequence[Image], im_out: Image):
```

Comment:
It might also be nice to specify that the Image objects passed in as arguments get modified in-place. When a function returns nothing, I expect it to have side-effects and/or mutate its arguments; but when a function has a return value, I don't expect this.

(Then again, maybe the use of the verb "pad" in "pad input and output images", rather than something like "returns a padded copy", implies this already?)

Reply:
Thank you for acknowledging the inconsistency in using type annotations + docstrings. It's definitely not something I follow a predetermined standard for, as I've seen myself get lazy ever since we stopped maintaining an API documentation page.

Reply:
I'm not sure I understand what you mean about the input arguments getting modified in-place, though?

Both output images are entirely new objects, AFAIU:

* `ims_in` -> `im_concat_list` -> `concat_data` (creates a new image object via `empty_like`) -> `im_concat`
* `im_out` -> `zeros_like` -> `im_out_padded`

Reply:
> Thank you for acknowledging the inconsistency in using type annotations + docstrings.

Sorry, I think I was unclear. I think it's fine to use both, just that they serve different purposes. The annotations tell us what each argument _is_, and the docstring tells us what each argument _means_. The _meaning_ is often clear from an argument name (or the function name, for the return value). But when returning multiple values, especially if some of them have the same type, it's nice to disambiguate them in the docstring.

> I'm not sure I understand what you mean about the input arguments getting modified in-place, though?

Sorry, I should have specified, I was referring to this line, which adds padding before the concatenation happens:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/9157b5956aacd26a8a947be84cf0cd3f7165f449/spinalcordtoolbox/image.py#L1753

I _think_ this mutates the `data` attribute of an `Image` object inside `ims_in` directly?

Reply:
Oh!! Good catch, gosh.

Reply:
> Sorry, I think I was unclear. 

Ah, I was unclear, too! I meant like, inconsistent in when I choose to add parameter descriptions in docstrings, and inconsistent about when I choose to specify the return type using `typing`. (I did neither in this case, hehe.)